tests/testsuite/tree.rs

```diff
@@ -1085,6 +1085,57 @@ fn duplicates_with_target() {
     p.cargo("tree -d --target=all").with_stdout("").run();
 }
 
+#[cargo_test]
+fn duplicates_with_proc_macro() {
+    Package::new("cat", "1.0.0").publish();
+    Package::new("cat", "2.0.0").publish();
+    Package::new("dep", "1.0.0")
+        .proc_macro(true)
+        .dep("cat", "1.0")
+        .publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "0.1.0"
+
+            [dependencies]
+            dep = "1.0"
+            cat = "2.0"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("tree")
```

Comment:
nit: would you be yup for splitting this commit into two, one commit that just adds tests, showing the existing behavior, with the second adding the fix and updating the tests?

As a reviewer this helps to show that this is indeed testing and fixing the right thing